chronos/achannels.nim

```diff
@@ -0,0 +1,254 @@
+#
+#    Chronos multi-threading asynchronous channels
+#             (c) Copyright 2018-Present
+#         Status Research & Development GmbH
+#
+#              Licensed under either of
+#  Apache License, version 2.0, (LICENSE-APACHEv2)
+#              MIT license (LICENSE-MIT)
+import strutils
+import asyncloop, handles, transport, asyncsync
+
+const hasThreadSupport* = compileOption("threads")
```

Comment:
Is this supposed to work at all without `--threads:on`? That would mean no locks whatsoever.

Reply:
Yes it supposed to work without `--threads:on` and in such case `Lock` will not be used, because there will be not possible to perform data race.